src/zac/core/views/utils.py

```diff
@@ -12,3 +20,69 @@ def get_zaak_from_query(request: HttpRequest, param: str = "zaak") -> Zaak:
 
     zaak = get_zaak(zaak_url=zaak_url)
     return zaak
+
+
+def check_document_permissions(document: Document, user: User):
+    """Check if a user has the permission to see a document"""
+    error_msg = "The user has insufficient permissions to access this document."
+
+    user_permissions = PermissionSet.objects.filter(
+        authorizationprofile__in=user.auth_profiles.all()
+    )
+
+    if not user_permissions.exists():
+        raise PermissionDenied(error_msg)
+
+    relevant_iot_permissions = InformatieobjecttypePermission.objects.filter(
+        catalogus=document.informatieobjecttype.catalogus
+    )
+
+    user_iot_permissions = relevant_iot_permissions.filter(
+        permission_set__in=user_permissions
+    )
+
+    if not user_iot_permissions.exists():
+        raise PermissionDenied(error_msg)
+
+    # The user has permissions to the catalog in question.
+    # Now need to check whether they have the right confidentiality level
+
+    document_va = VertrouwelijkheidsAanduidingen.get_choice(
+        document.informatieobjecttype.vertrouwelijkheidaanduiding
```

Comment:
```suggestion
        document.vertrouwelijkheidaanduiding
```
The document itself has a VA - which is by default the value from the IOT, but may be overridden.